html-to-docx2/attempts/attempt_2.py

```
@@ -0,0 +1,620 @@
+            # Blocks exist in both – but if paragraph text is equal, we still check
+            # for formatting-only changes and record them via rPrChange.
+            for (oe, ok), (ne, nk) in zip(old_slice, new_slice):
+                if ok == 'p' and nk == 'p':
+                    body.append(build_paragraph_with_diffs(oe, ne, author, date_iso, cidgen))
+                else:
+                    # Copy new block (tables/others)
+                    body.append(deepcopy(ne))
+
+        elif tag == 'delete':
+            # Entire blocks removed => emit <w:del> wrappers (block-level)
+            for oe, ok in old_slice:
```

Ruff (0.15.2)

[warning] Loop control variable `ok` not used within loop body

(B007)